tests/gadget-zero/usb-gadget0.c

```diff
@@ -181,37 +184,60 @@ static struct {
 	.pattern_counter = 0,
 };
 
+/* aligned / unaligned test flag.
+ * Using a bool type for this instead, in some cases, can cause gcc to place other critical buffers
```

Comment:
That's what caused my problems. I had a global bool, that gcc allocated as a uint8 , and in certain conditions other uint8_t arrays were allocated right after, with no padding, causing an unaligned control buffer and associated issues.